llvm/lib/Support/Mustache.cpp

```diff
@@ -681,10 +685,59 @@ void ASTNode::renderChild(const json::Value &Contexts, llvm::raw_ostream &OS) {
     Child->render(Contexts, OS);
 }
 
+void ASTNode::indentTextNode(std::string &Body, size_t Indentation,
+                             bool FinalNode) {
+  std::string spaces(Indentation, ' ');
+  size_t pos = 0;
+  size_t LastChar = std::string::npos;
+
+  if (FinalNode)
+    // body.erase(body.find_last_not_of(" \t\r\f\v") + 1);
+    LastChar = Body.find_last_not_of(" \t\r\f\v");
+
+  while ((pos = Body.find('\n', pos)) != std::string::npos) {
+    if ((!FinalNode) || (pos != LastChar)) {
+      Body.insert(pos + 1, spaces);
+      pos += 1 + Indentation;
+    } else {
+      break;
+    }
+  }
+}
```

Comment:
```suggestion
  while ((pos = Body.find('\n', pos)) != std::string::npos && !(FinalNode && pos == LastChar)) {
      Body.insert(pos + 1, spaces);
      pos += 1 + Indentation;
  } 
```

In any case you can use an early break to avoid the need for an else if you invert the `if` condition, so even if you don't want to combine them into the loop condition, there's a nicer way to write it.

```suggestion
  while ((pos = Body.find('\n', pos)) != std::string::npos) {
    if (FinalNode && (pos == LastChar))
      break;
     
    Body.insert(pos + 1, spaces);
    pos += 1 + Indentation;
  }
}
```